x/harvest/keeper/borrow.go

```diff
@@ -0,0 +1,138 @@
+	} else {
+		locBorrowMoneyMarket, found := k.GetMoneyMarket(ctx, amount.Denom)
+		if !found {
+			return sdkerrors.Wrapf(types.ErrMarketNotFound, "no market found for denom %s", amount.Denom)
+		}
+		locBorrowAssetPrice, err := k.pricefeedKeeper.GetCurrentPrice(ctx, locBorrowMoneyMarket.SpotMarketID)
+		if err != nil {
+			return sdkerrors.Wrapf(types.ErrPriceNotFound, "1: no price found for market %s", locBorrowMoneyMarket.SpotMarketID)
+		}
+		borrowMoneyMarket = locBorrowMoneyMarket
+		borrowAssetPrice = locBorrowAssetPrice.Price
+		borrowUSDValue = sdk.NewDecFromInt(amount.Amount).Mul(borrowAssetPrice)
+	}
+
+	// Get the user's deposits
+	deposits := k.GetDepositsByUser(ctx, borrower)
+	if len(deposits) == 0 {
+		return sdkerrors.Wrapf(types.ErrDepositsNotFound, "no deposits found for %s", borrower)
+	}
+
+	// Get the total value of the user's deposits
+	totalValueDeposits := sdk.ZeroDec()
+	for _, deposit := range deposits {
+		if deposit.Amount.Denom == USDX {
+			totalValueDeposits = totalValueDeposits.Add(sdk.NewDecFromInt(deposit.Amount.Amount))
+		} else {
+			depositMoneyMarket, found := k.GetMoneyMarket(ctx, deposit.Amount.Denom)
+			if !found {
+				return sdkerrors.Wrapf(types.ErrMarketNotFound, "no market found for denom %s", deposit.Amount.Denom)
+			}
+			depositAssetPrice, err := k.pricefeedKeeper.GetCurrentPrice(ctx, depositMoneyMarket.SpotMarketID)
+			if err != nil {
+				return sdkerrors.Wrapf(types.ErrPriceNotFound, "2: no price found for market %s", depositMoneyMarket)
+			}
+			depositUSDValue := sdk.NewDecFromInt(deposit.Amount.Amount).Mul(depositAssetPrice.Price)
+			totalValueDeposits = totalValueDeposits.Add(depositUSDValue)
+		}
+	}
+
+	// Get the total value of the user's borrows
+	borrows := k.GetBorrowsByUser(ctx, borrower)
+	totalValueBorrows := sdk.ZeroDec()
+	for _, borrow := range borrows {
+		if borrow.Amount.Denom == USDX {
+			totalValueBorrows = totalValueBorrows.Add(sdk.NewDecFromInt(borrow.Amount.Amount))
+		} else {
+			borrowAssetPrice, err := k.pricefeedKeeper.GetCurrentPrice(ctx, borrowMoneyMarket.SpotMarketID)
+			if err != nil {
+				return err
+			}
+			borrowUSDValue := sdk.NewDecFromInt(borrow.Amount.Amount).Mul(borrowAssetPrice.Price)
+			totalValueBorrows = totalValueBorrows.Add(borrowUSDValue)
+		}
+	}
+
+	// Value of borrow cannot be greater than:
+	// (total value of user's deposits * the borrow asset denom's LTV ratio) - funds already borrowed
+	borrowValueLimit := totalValueDeposits.Mul(borrowMoneyMarket.BorrowLimit.LoanToValue).Sub(totalValueBorrows)
+	if borrowUSDValue.GT(borrowValueLimit) {
+		return sdkerrors.Wrapf(types.ErrInsufficientLoanToValue,
+			"requested borrow %s is greater than maximum valid borrow %s",
+			amount, sdk.NewCoin(amount.Denom, borrowValueLimit.Quo(borrowAssetPrice).TruncateInt()))
+	}
+
+	return nil
+}
+
+func (k Keeper) validateBorrowSystem(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
```

Comment:
delete this function

Reply:
Deleted in https://github.com/Kava-Labs/kava/pull/702/commits/277c4b1936981182fc09ae0c6ea0973d9cbbca8a